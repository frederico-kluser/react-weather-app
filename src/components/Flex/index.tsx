import Container from './styled';

interface FlexInterface {
  children: any;
  row?: boolean;
}

const Flex = ({children, row}: FlexInterface) => (
  <Container row={row}>{children}</Container>
);

export default Flex;
