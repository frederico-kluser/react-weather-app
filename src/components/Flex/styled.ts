import styled from 'styled-components';
import { mediaQuery } from '../../utils/style';

interface ContainerInterface {
  row?: boolean;
}

const Container = styled.div<ContainerInterface>`
  justify-content: center;
  display: flex;
  flex-direction: ${({row}) => row ? 'row' : 'column'};
  ${mediaQuery(`
    flex-direction: column;
    margin-bottom: 25px;
  `)}
`;

export default Container;
