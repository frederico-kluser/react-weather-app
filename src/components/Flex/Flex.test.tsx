import renderer from 'react-test-renderer';
import Flex from '.';

test('Flex Renderer', () => {
  const component = renderer.create(
    <Flex><></></Flex>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});