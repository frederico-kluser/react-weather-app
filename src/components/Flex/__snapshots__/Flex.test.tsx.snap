// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Renderer 1`] = `
<div
  className="sc-bdvvtL ccbulY"
/>
`;
